fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -2450,6 +2519,11 @@ private LoanScheduleDTO rescheduleNextInstallments(final MathContext mc, final L
 
         BigDecimal rescheuleInterestPortionTobeAppropriated = rescheuleInterestPortionTotal.subtract(rescheuleInterestPortionTobeRetained);
         loanApplicationTerms.setInterestTobeApproppriated(Money.of(loan.getCurrency(), rescheuleInterestPortionTobeAppropriated));
+
+        loanScheduleParams = null;
```

Comment:
add a comment that will remind us to figure out why clearing the terms is necessary